### DISCUSSION TITLE
"BG reading unavailable" - companion app on Pixel 8

### DISCUSSION THREAD

**Discussion by Shhlink** [NightscoutFoundation/xDrip#3711] [Q&A]
Hi everyone

I am trying to set up xdrip on my pixel 8 so that I can start reading my blood sugars from my dexcom onto my Garmin smart watch using a custom watch face. 

When setting up xdrip on my phone, using companion app as the data source, I receive the follow error in my notification bar:
"BG reading unavailable
xdrip data collection service is running"

I have made sure to use the following settings but any other help would be greatly appreciated:

- hardware data source set to companion 
- companion Bluetooth on
- all notifications are on and I have allowed the app to read my notifications 
- battery in android settings is set to unrestricted 

I have also made sure to stop and start a new sensor within the xdrip app. 

Unfortunately after reading through the docs and a few other threads I am still unsure what to do from here and I'd greatly appreciate some help to set xdrip up.

Thanks all

**Comment by Navid200** [comment]
Identify the companion app.
Post a screenshot of the notification from the companion app.
Post a screenshot of the xDrip classic status page.

**Reply by Shhlink** [reply]
Thanks for responding navid. Just to be clear, the companion app is a setting within xdrip, it's not a separate app right? 
![Screenshot_20241011-141427](https://github.com/user-attachments/assets/92f5c0ca-b7c7-40d2-8a97-6babf9b0230b)
![Screenshot_20241011-141439](https://github.com/user-attachments/assets/486988d3-e92d-436e-bd7a-ae8556bc2a33)
![Screenshot_20241011-141525](https://github.com/user-attachments/assets/4d1c82b6-29e6-4ded-b96a-c0b9139a7f26)

**Reply by Navid200** [reply]
There is a list of apps you can run on your phone as well as xDrip that xDrip can get readings from.  They need to run on the same phone.   xDrip is set to companion app to reflect the concept, getting readings from a companion app.

As an example, Dexcom G6 app is on the list.

There is a Dexcom app icon on your phone.  What is it, G6 or what?

**Reply by Navid200** [reply]
Have you followed this?  https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html

**Reply by Shhlink** [reply]
Ok thought that was the case. Yes I have the dexcom g6 app on my phone and it's running fine, with notifications coming through (as you can see in the screenshot).
Looking through that document you just sent through I can confirm that I have all those same settings including the companion app Bluetooth settings

**Reply by Navid200** [reply]
What do the xDrip logs show?

**Comment by Shhlink** [comment]
Is this what you mean? 
![Uploading Screenshot_20241011-144216.png…]()

**Reply by Shhlink** [reply]
![Screenshot_20241011-144216](https://github.com/user-attachments/assets/e6d2af7a-739d-4249-9e89-5f79bc08c7c2)

**Reply by Navid200** [reply]
I expected to see logs from a different method.  
Have you restarted the phone?

**Reply by Shhlink** [reply]
I have just restarted the phone and here are the logs again. Let me know if I am screenshotting the wrong thing? 
![Screenshot_20241011-150532](https://github.com/user-attachments/assets/cdf156b1-4353-4265-b1b7-714717a309bc)

**Reply by Navid200** [reply]
Why does the Dexcom app notification have no time in it?

**Reply by Shhlink** [reply]
The dexcom app shows a continuous notification in the "quick glance" view. Is that wrong?

**Reply by Shhlink** [reply]
![Screenshot_20241011-151622](https://github.com/user-attachments/assets/b787a1d4-9f4d-4b3f-aee8-1dae81c79237)

**Reply by Navid200** [reply]
@Shhlink Can you update to the latest Nightly?

It would be this: https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.09.27

**Reply by Shhlink** [reply]
I have updated to the latest nightly and still no luck. See attached 
![Screenshot_20241011-153147](https://github.com/user-attachments/assets/01870d4a-2cac-4f8a-b3e5-4ec13c862867)
![Screenshot_20241011-153214](https://github.com/user-attachments/assets/84d7af6d-c767-4d30-835b-9cf1dc8ba936)

**Comment by Navid200** [comment]
@jamorham Why are there no logs from the UiBasedCollector method?

What is causing this behavior?

**Comment by jamorham** [comment]
@Shhlink can you confirm which country/region you are in? Also please check that xDrip has notification access. You would have had to go through the process of enabling restricted settings and then enabling it.

**Reply by Shhlink** [reply]
Yes, I have enabled that setting in the android settings. I am located in Australia

**Reply by Navid200** [reply]
I don't understand why there are no logs.  Let's enable detail logs then:

Please go to the following and enter `UiBasedCollector:v`  
  
`Settings` > `Less common settings` > `Extra Logging Settings` > `Extra tags for logging` 

When everything is working again, to disable, go back and delete to empty and enter.

**Reply by Shhlink** [reply]
Not sure if I did that correctly but here's a screenshot
![Screenshot_20241012-043725](https://github.com/user-attachments/assets/8d92b16e-f28b-45d9-9466-b50a7547d8d0)

**Reply by Navid200** [reply]
Do you see any more logs with that heading UiBasedCollector?

**Reply by Shhlink** [reply]
Yeh im getting alot of missed reading logs "aggressively restarting collector service due to missed reading"

**Reply by Navid200** [reply]
A new xDrip release will come out in the next few minutes to hours.  
When it comes out, please try it.  Hopefully, it will fix it.

**Reply by Shhlink** [reply]
Awesome thanks so much. Is that the nightly version?

**Reply by Navid200** [reply]
Yes, All xDrip new releases come out as Nightly releases.

**Reply by Navid200** [reply]
It has been released.  Please try when you can.

**Reply by Navid200** [reply]
https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.10.11

**Reply by Shhlink** [reply]
Thanks. I am no receiving sugar readings within the app but nothing in my notifications bar. Is there a setting I should change? Also, once it is working in my notifications bar, do I need to make sure my dexcom is also showing notifications ? (I.e. I'll always have double readings, one from dexcom and the other from xdrip?)
![Screenshot_20241012-092319](https://github.com/user-attachments/assets/085fd8fe-7d9e-48e2-9ec0-b67b0570c99c)
![Screenshot_20241012-092321](https://github.com/user-attachments/assets/95831b75-2141-4614-864c-e2f9f8dcba55)

**Reply by Navid200** [reply]
There should be no double readings.

**Reply by Navid200** [reply]
The companion app should be set to show notifications for readings to appear in xDrip.  
Companion Bluetooth should be enabled so that xDrip can get backfill.

**Comment by Shhlink** [comment]
Fantastic. Looks like it's all working! So does that mean I can turn off "glance view" from the dexcom app?
![Uploading Screenshot_20241012-093257.png…]()

**Reply by Shhlink** [reply]
![Screenshot_20241012-093257](https://github.com/user-attachments/assets/1519c2f3-3c44-4949-af2f-9104cd54c3ff)

**Reply by Navid200** [reply]
I believe if you turn off glance view, Dexcom app will stop showing a new notification for each reading and xDrip will stop showing readings.  You can give it a try.

**Reply by Navid200** [reply]
This is just for info and not related to your questions.  Just in case may be useful: https://navid200.github.io/xDrip/docs/Dexcom/Clarity.html

**Reply by Shhlink** [reply]
It looks to be working without glance view for the dexcom app. Will respond here if it stops working for anyone else in the future !

**Comment by Shhlink** [comment]
Yes okay. In that case I might turn off glance view for xdrip as I'd like to just use xdrip to send readings to my watch. Will that work with xdrip?

**Reply by Navid200** [reply]
I'm not 100% sure what you are asking here.
I included, in my last post, a page explaining pros and cons of using xDrip as a collector.  You currently are not using xDrip as a collector.
Are you asking about that, or are you asking about notifications still?

**Reply by Navid200** [reply]
If you use xDrip as a collector, you will have to uninstall Dexcom app.  It will be a different approach.  
As far as using xDrip with a watch through WearOS, it makes no difference how xDrip gets its readings.

**Reply by Navid200** [reply]
This is how you can use xDrip as a collector with G6:
https://navid200.github.io/xDrip/docs/Starting-G6.html

**Reply by Shhlink** [reply]
Thanks for all your help navid. I was just asking if I would still receive blood sugar readings if I turned off glance view on the dexcom to stop duplicate readings (i.e. Two apps giving me the same information) I have since discovered that it doesn't matter. Thanks very much for assisting me in setting up my xdrip and for sharing the pros/cons list.

**Reply by Navid200** [reply]
You're welcome although it was JamOrHam who did the work that fixed the malfunction.
